app/models/study.rb

```diff
@@ -1132,6 +1132,22 @@ def unique_genes
     Gene.where(study_id: self.id, :study_file_id.in => self.expression_matrix_files.map(&:id)).pluck(:name).uniq
   end
 
+  # Get species scientific name for a searched gene.  E.g. PTEN -> Homo sapiens.
+  def infer_species(searched_gene)
+    # Get gene record for searched gene string
+    gene = self.genes.by_name_or_id(searched_gene, self.expression_matrix_files.map(&:id))
+
+    # Get matrices containing this gene.  (Same gene name can be in multiple matrices.)
+    study_file_ids = Gene.where(study_id: self.id, name: gene['name']).pluck(:study_file_id)
```

Comment:
```suggestion
    study_file_ids = genes.where(name: gene['name']).pluck(:study_file_id)
```

this gets the same thing, but will be able to take advantage of rails caching if available

Reply:
Actually, in this case this is incorrect.  Calling `genes` (which is an alias for `self.genes`) will return a `Mongoid::Relations::Targets::Enumerable`, which eager-loads all `Gene` documents for a given study.  The performance on this is actually quite poor.  Calling `Gene.where(...)` simply returns a `Mongoid::Criteria` which will lazy-load any documents matching the query and is vastly more performant.